core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala

```diff
@@ -243,4 +244,59 @@ class WorkerSuite extends SparkFunSuite with Matchers with BeforeAndAfter {
       ExecutorStateChanged("app1", 0, ExecutorState.EXITED, None, None))
     assert(cleanupCalled.get() == value)
   }
+
+  test("removes our metadata of all executors registered for the given " +
+    "application after WorkDirCleanup when config spark.shuffle.service.db.enabled=true") {
+    testWorkDirCleanupAndRemoveMetadataWithConfig(true)
+  }
+
+  test("don't removes our metadata of all executors registered for the given " +
+    "application after WorkDirCleanup when config spark.shuffle.service.db.enabled=false") {
```

Comment:
minor -- as I mentioned below, both of these tests should still test dir cleanup, so maybe we can rename a little to something like

WorkDirCleanup cleans app dirs and shuffle metadata when spark.shuffle.service.db.enabled=true

WorkdDirCleanup cleans only app dirs when spark.shuffle.service.db.enabled=false